src/AppBundle/Service/OccurrenceSplitterService.php

```diff
@@ -0,0 +1,179 @@
+    public function __construct(string $dateSeparatorTime, string $dateSeparatorTimezone)
+    {
+        $exploded = explode(':', $dateSeparatorTime);
+        $this->splitHour = (int) $exploded[0];
+        $this->splitMinute = (int) $exploded[1];
+        $this->splitSecond = (int) $exploded[2];
+        $this->dateSeparatorTimezone = new \DateTimeZone($dateSeparatorTimezone);
+
+        $propertyInfo = new ReflectionExtractor();
+        $this->occurrenceTraitProperties = $propertyInfo->getProperties(OccurrenceTrait::class);
+
+        $this->propertyAccessor = PropertyAccess::createPropertyAccessor();
+    }
+
+    /**
+     * Get new DailyOccurrences from an Occurrence.
+     *
+     * @param Occurrence $occurrence
+     *
+     * @throws \Exception
+     *
+     * @return Collection
+     */
+    public function getDailyOccurrences(Occurrence $occurrence): Collection
+    {
+        $dailyOccurrences = new ArrayCollection();
+        if ($occurrence->getStartDate() && $occurrence->getEndDate() && $occurrence->getStartDate() <= $occurrence->getEndDate()) {
+            $this->createDailyOccurrenceCollection($occurrence, $dailyOccurrences);
+        }
+
+        return $dailyOccurrences;
+    }
+
+    /**
+     * Copy values of OccurrenceTrait properties from one DailyOccurrence to another.
+     *
+     * @param DailyOccurrence $to
+     * @param DailyOccurrence $from
+     */
+    public function copyOccurrenceTraitPropertyValues(DailyOccurrence $to, DailyOccurrence $from): void
+    {
+        foreach ($this->occurrenceTraitProperties as $propertyPath) {
+            $value = $this->propertyAccessor->getValue($from, $propertyPath);
+            $this->propertyAccessor->setValue($to, $propertyPath, $value);
+        }
+    }
+
+    /**
+     * Create a collection of DailyOccurrences from an Occurrence.
+     *
+     * @param Occurrence $occurrence
+     * @param Collection $dailyOccurrences
+     *
+     * @throws \Exception
+     *
+     * @return Collection
+     */
+    private function createDailyOccurrenceCollection(Occurrence $occurrence, Collection $dailyOccurrences): Collection
+    {
+        $tempOccurrence = clone $occurrence;
+        $splitDate = $this->getFirstSplitDateTime($tempOccurrence->getStartDate());
+
+        if ($tempOccurrence->getStartDate() < $splitDate && $tempOccurrence->getEndDate() > $splitDate) {
+            while ($tempOccurrence->getStartDate() < $splitDate && $tempOccurrence->getEndDate() > $splitDate) {
+                $dailyOccurrence = $this->createDailyOccurrence($tempOccurrence->getStartDate(), $splitDate, $occurrence);
+                $dailyOccurrences->add($dailyOccurrence);
+
+                $tempOccurrence->setStartDate($splitDate);
+                $splitDate = $this->getFirstSplitDateTime($tempOccurrence->getStartDate());
+            }
+        }
+
+        $dailyOccurrence = $this->createDailyOccurrence($tempOccurrence->getStartDate(), $tempOccurrence->getEndDate(), $occurrence);
```

Comment:
I'm not sure, but I have a suspicion that we may add an empty `DailyOccurrence` (with `startDate === endDate`) here in some edge cases. Maybe add a check for `$tempOccurrence->getStartDate() < $tempOccurrence->getEndDate()`?

Reply:
Fixed